pkg/cmd/roachprod/flags.go

```diff
@@ -131,7 +133,7 @@ func initFlags() {
 			providerOptsContainer[providerName].ConfigureCreateFlags(createCmd.Flags())
 
 			for _, cmd := range []*cobra.Command{
-				destroyCmd, extendCmd, listCmd, syncCmd, gcCmd,
+				destroyCmd, extendCmd, listCmd, syncCmd, gcCmd, setupSSHCmd, startCmd, pgurlCmd, adminurlCmd,
```

Comment:
Methinks only `setupSSHCmd` calls `Sync`, which in turn may invoke `SyncDNS`. The others don't need parameterized DNS flags.